src/Services/UrlService.php

```diff
@@ -233,4 +249,38 @@ public function redirectTo($redirectURL)
 
         return pluginApp(Response::class)->redirectTo($redirectURL);
     }
+
+    /**
+     * Takes a key and generates a link for it based on current state
+     *
+     * @param $internalLink
+     * @param array $varargs
+     * @return string
+     */
+    public function getInternalLink($internalLink, $envargs = [])
+    {
+        $internalUrlBuilder = pluginApp(InternalUrlBuilder::class);
+        $shopUrls = pluginApp(ShopUrls::class);
+        $generatedURL = "";
+
+        if(!is_array($envargs)) {
+            return $generatedURL;
+        }
+
+        switch($internalLink) {
+            case "retoure":
+                $generatedURL = $internalUrlBuilder->buildRetoureUrl($envargs['order']['id']);
```

Comment:
Auch hier ist für außenstehende nicht verständlich, warum ein Array mit ['order']['id'] übergeben werden sollte, wenn nur die ID selbst in der URL landet. Sinnvoller wäre es, die Varargs 1:1 durchzureichen:
```
$generatedURL = $shopUrls->returns($varargs[0], $varargs[1])
```
Damit wäre der Aufruf `getInternalLink("retoure", order.id, accessKey)`